File: Java/dijkstra/llama-3.2-90b-vision-preview/Llama_3_2_90b_vision_preview.java
import java.util.*;

class Node implements Comparable<Node> {
    int id;
    int distance;

    public Node(int id, int distance) {
        this.id = id;
        this.distance = distance;
    }

    @Override
    public int compareTo(Node other) {
        return Integer.compare(this.distance, other.distance);
    }
}

public class Llama_3_2_90b_vision_preview {
    private static final int INF = Integer.MAX_VALUE;

    public static void dijkstra(int[][] graph, int start) {
        int numNodes = graph.length;
        int[] distances = new int[numNodes];
        boolean[] visited = new boolean[numNodes];

        // Initialize distances and visited arrays
        for (int i = 0; i < numNodes; i++) {
            distances[i] = INF;
            visited[i] = false;
        }

        distances[start] = 0; // Distance to start node is 0

        PriorityQueue<Node> queue = new PriorityQueue<>();
        queue.add(new Node(start, 0));

        while (!queue.isEmpty()) {
            Node node = queue.poll();
            int nodeId = node.id;

            if (visited[nodeId]) {
                continue;
            }

            visited[nodeId] = true;

            for (int neighbor = 0; neighbor < numNodes; neighbor++) {
                if (graph[nodeId][neighbor] > 0 && !visited[neighbor]) {
                    int newDistance = distances[nodeId] + graph[nodeId][neighbor];

                    if (newDistance < distances[neighbor]) {
                        distances[neighbor] = newDistance;
                        queue.add(new Node(neighbor, newDistance));
                    }
                }
            }
        }

        // Print shortest distances
        System.out.println("Shortest distances from node " + start + ":");
        for (int i = 0; i < numNodes; i++) {
            System.out.println("Node " + i + ": " + distances[i]);
        }
    }

    public static void main(String[] args) {
        // Número de nós
        int numNodes = 1000;

        // Gerador de números aleatórios
        Random random = new Random();

        // Matriz de adjacência para o grafo
        int[][] graph = new int[numNodes][numNodes];

        // Criando arestas aleatórias
        for (int i = 0; i < numNodes; i++) {
            for (int j = i + 1; j < numNodes; j++) {
                if (random.nextDouble() < 0.05) { // 5% de chance de criar uma aresta
                    int weight = random.nextInt(10) + 1;  // Peso aleatório entre 1 e 10
                    graph[i][j] = weight;
                    graph[j][i] = weight;  // Grafo não direcionado
                }
            }
        }

        // Nó de partida aleatório
        int startNode = random.nextInt(numNodes);

        // Calculando as distâncias usando o algoritmo de Dijkstra
        dijkstra(graph, startNode);
    }
}

File: Python/dijkstra/llama-3.2-90b-vision-preview/llama-3.2-90b-vision-preview.py
import heapq
import random
class Graph:
    def __init__(self):
        """
        Initialize an empty graph.
        """
        self.nodes = set()
        self.edges = {}

    def add_node(self, node):
        """
        Add a node to the graph.

        Args:
            node (str): The node to add.
        """
        self.nodes.add(node)
        if node not in self.edges:
            self.edges[node] = []

    def add_edge(self, from_node, to_node, weight):
        """
        Add an edge to the graph.

        Args:
            from_node (str): The node where the edge starts.
            to_node (str): The node where the edge ends.
            weight (int): The weight of the edge.
        """
        self.edges[from_node].append((to_node, weight))

    def dijkstra(self, start_node):
        """
        Run Dijkstra's algorithm to find the shortest paths from the start node.

        Args:
            start_node (str): The node where the algorithm starts.

        Returns:
            distances (dict): A dictionary with the shortest distances from the start node to all other nodes.
            previous_nodes (dict): A dictionary with the previous node in the shortest path from the start node to all other nodes.
        """
        distances = {node: float('infinity') for node in self.nodes}
        distances[start_node] = 0
        previous_nodes = {node: None for node in self.nodes}

        priority_queue = [(0, start_node)]

        while priority_queue:
            current_distance, current_node = heapq.heappop(priority_queue)

            if current_distance > distances[current_node]:
                continue

            for neighbor, weight in self.edges[current_node]:
                distance = current_distance + weight

                if distance < distances[neighbor]:
                    distances[neighbor] = distance
                    previous_nodes[neighbor] = current_node
                    heapq.heappush(priority_queue, (distance, neighbor))

        return distances, previous_nodes

    def get_shortest_path(self, start_node, end_node):
        """
        Get the shortest path from the start node to the end node.

        Args:
            start_node (str): The node where the path starts.
            end_node (str): The node where the path ends.

        Returns:
            path (list): A list of nodes representing the shortest path from the start node to the end node.
        """
        distances, previous_nodes = self.dijkstra(start_node)
        path = []
        current_node = end_node

        while current_node is not None:
            path.append(current_node)
            current_node = previous_nodes[current_node]

        path.reverse()
        return path


def main():
    graph = Graph()

    # Criar 1000 nós numerados de 0 a 999
    for i in range(1000):
        graph.add_node(str(i))

    # Adicionar arestas aleatórias entre os nós com pesos aleatórios
    for _ in range(1000):  # Criar 3000 conexões aleatórias
        from_node = str(random.randint(0, 999))
        to_node = str(random.randint(0, 999))
        weight = random.randint(1, 100)  # Peso entre 1 e 100
        if from_node != to_node:  # Evita laços
            graph.add_edge(from_node, to_node, weight)

    # Definir um nó inicial e executar Dijkstra
    start_node = '0'
    distances, _ = graph.dijkstra(start_node)

if __name__ == "__main__":
    main()